gofakes3.go

```diff
@@ -618,7 +618,15 @@ func (g *GoFakeS3) createObject(bucket, object string, w http.ResponseWriter, r
 		return err
 	}
 
-	result, err := g.storage.PutObject(bucket, object, meta, rdr, size)
+	if r.Header.Get("x-amz-storage-class") != "" {
+		if ok := IsStorageClassValid(r.Header.Get("x-amz-storage-class")); !ok {
+			return ErrInvalidStorageClass
+		}
+	} else {
```

Comment:
Why do you validate the storage class? There're several s3-compatible object storage services that have diffent storage class names.

Reply:
done.